### PR TITLE
various: transition formulae over to gnupg2

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Birthed out of https://github.com/Homebrew/homebrew-core/pull/2250, with various design tweaks.

Depends on https://github.com/Homebrew/brew/pull/675.
Depends on https://github.com/Homebrew/brew/pull/676.

### CODE REVIEW

File: Formula/zero-install.rb [hunk 1]
```diff
@@ -14,77 +23,75 @@ class ZeroInstall < Formula
...
   depends_on "pkg-config" => :build
   depends_on "ocaml" => :build
+  depends_on "ocamlbuild" => :build
   depends_on "opam" => :build
+  depends_on "camlp4" => :build
   depends_on "gtk+" => :optional
-  depends_on :gpg
-
-  # 0install.2.8 resources from https://github.com/ocaml/opam-repository.git
+  depends_on :gpg => :run
```
Comment (MikeMcQuaid) [inline Formula/zero-install.rb]: Is this always `:run` now?

**Comment by MikeMcQuaid** [comment]
LGTM 👍

File: Formula/zero-install.rb (hunk 1 above)
Comment (DomT4) [inline Formula/zero-install.rb]: Here specifically or `:gpg` in general? It's only ever really needed during build if upstream has `make check`, etc using `gpg` to test against, or `Makefile`s, etc are checking `gpg` exists before proceeding with installation. Both cases are pretty rare.

Comment (MikeMcQuaid) [inline Formula/zero-install.rb]: In general. As the answer is "needed sometimes": 👍 to keep as-is.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Pull request closed without merging**